server/app/assets/javascripts/admin_programs.ts

```diff
@@ -24,88 +24,124 @@ class AdminPrograms {
     addEventListenerToElements('#common-intake-checkbox', 'click', () => {
       const commonIntakeCheckbox = <HTMLInputElement>(
         document.querySelector('#common-intake-checkbox')
       )
 
-      const programCategoryCheckboxes = document.querySelectorAll(
-        '[id^="checkbox-category"]',
+      // Program categories
+      this.updateUSWDSCheckboxesDisabledState(
+        /* fieldSelectors= */ '[id^="checkbox-category"]',
+        /* shouldDisable= */ commonIntakeCheckbox.checked,
       )
-      programCategoryCheckboxes.forEach((checkbox) => {
-        const category = checkbox as HTMLInputElement
-        if (commonIntakeCheckbox.checked) {
-          category.disabled = true
-          category.checked = false
-        } else {
-          category.disabled = false
-        }
-      })
 
+      // Long program description
       const longDescription = document.getElementById(
         'program-display-description-textarea',
       ) as HTMLInputElement
       const northStarUiEnabled =
         longDescription.dataset.northstarEnabled === 'true'
-      this.maybeDisableField(
-        longDescription,
-        commonIntakeCheckbox.checked && northStarUiEnabled,
+      this.updateTextFieldDisableState(
+        /* fieldElement= */ longDescription,
+        /* shouldDisable= */ commonIntakeCheckbox.checked && northStarUiEnabled,
       )
 
-      const applicationStepTitles = document.querySelectorAll(
-        'input[id^="apply-step"]',
-      )
-      const applicationStepDescriptions = document.querySelectorAll(
-        'textarea[id^="apply-step"]',
+      // Application steps
+      this.updateTextFieldsDisabledState(
+        /* fieldSelectors= */ 'input[id^="apply-step"]',
+        /* shouldDisable= */ commonIntakeCheckbox.checked,
       )
-      this.maybeDisableApplicationSteps(
-        applicationStepTitles,
-        commonIntakeCheckbox,
+      this.updateTextFieldsDisabledState(
+        /* fieldSelectors= */ 'textarea[id^="apply-step"]',
+        /* shouldDisable= */ commonIntakeCheckbox.checked,
       )
-      this.maybeDisableApplicationSteps(
-        applicationStepDescriptions,
-        commonIntakeCheckbox,
-      )
-      // remove the required indicator from the first application step
+      // Show the required indicator in the first application step only when
+      // the application steps are enabled
       const applicationStepOneDiv = document.querySelector('#apply-step-1-div')
-      if (commonIntakeCheckbox.checked) {
-        const requiredIndicators =
-          applicationStepOneDiv?.querySelectorAll('span')
-        requiredIndicators?.forEach((indicator) => {
+      const requiredIndicators = applicationStepOneDiv?.querySelectorAll('span')
+      requiredIndicators?.forEach((indicator) => {
+        if (commonIntakeCheckbox.checked) {
           indicator.classList.add('hidden')
-        })
-      }
+        } else {
+          indicator.classList.remove('hidden')
```

Comment:
Technically, we should add 'optional' to match the styling of other fields.
However, that adds more complexity since we would need to change the actual text. I think this is fine, since not having '*' implies optional

(separately, wish we didn't have the word optional since not having '*' implies that.. but that is a separate conversation that can be addressed with the admin ui redesign)

Reply:
I 100% agree with you, it's fine to not add "optional" in and I would love to get rid of that text entirely when we re-do this page.